cfme/tests/automate/custom_button/test_infra_objects.py

```diff
@@ -339,3 +380,59 @@ def test_custom_button_expression(appliance, request, setup_obj, button_group, e
             assert not custom_button_group.is_displayed
             setup_obj.add_tag(tag)
             assert custom_button_group.is_displayed
+
+
+@pytest.mark.uncollectif(lambda button_group: "VM_INSTANCE" not in button_group)
+def test_open_url(request, setup_obj, button_group, method):
+    """ Test Open url functionality of custom button.
+
+    Polarion:
+        assignee: ndhandre
+        initialEstimate: 1/2
+        caseimportance: high
+        testSteps:
+            1. Appliance with Infra provider
+            2. Create ruby method for url functionality
+            3. Create custom button group with the Object type
+            4. Create a custom button with open_url option and respective method
+            5. Navigate to object Detail page
+            6. Execute custom button
+            7. Check new tab open or not with respective url
+    """
+
+    group, obj_type = button_group
+    button = group.buttons.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        open_url=True,
+        display_for="Single entity",
+        system="Request",
+        request=method.name,
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    view = navigate_to(setup_obj, "Details")
+    custom_button_group = Dropdown(view, group.hover)
+    assert custom_button_group.has_item(button.text)
+
+    # TODO: Move windows handling functionality to browser
+    initial_count = len(view.browser.selenium.window_handles)
+    main_window = view.browser.selenium.current_window_handle
+    custom_button_group.item_select(button.text)
+
+    wait_for(
+        lambda: len(view.browser.selenium.window_handles) > initial_count,
+        timeout=120,
+        message="Check for window open",
+    )
+    open_url_window = set(view.browser.selenium.window_handles) ^ {main_window}
```

Comment:
REQUIRED: it works but this is wrong operation.  you use symmetric difference instead of (difference) subtract.